source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js

```diff
@@ -0,0 +1,301 @@
+/**
+ * @jest-environment jsdom
+ */
+const JsCookie = require("js-cookie");
+const testCookiePageTemplate = require("./test-cookie-page-template");
+
+const formInputsForMock = jest.fn((param) => {
+  // Return the form input objects
+  let inputs = [];
+  if (param === "on") {
+    inputs = [{ checked: true, value: "on" }, { value: "off" }];
+  } else if (param === "off") {
+    inputs = [{ checked: true, value: "off" }, { value: "on" }];
+  } else if (param === "neither") {
+    inputs = [{ value: "on" }, { value: "off" }];
+  }
+  return inputs;
+});
+
+const mockSubmitEvent = jest.fn((param) => {
+  const submitEvent = jest.fn();
+  submitEvent.target = jest.fn();
+  submitEvent.preventDefault = jest.fn();
+  submitEvent.target.querySelectorAll = jest.fn();
+  submitEvent.target.querySelectorAll.mockImplementation(() => {
+    return formInputsForMock(param);
+  });
+  return submitEvent;
+});
+
+describe("Cookie settings", () => {
+  beforeEach(() => {
+    window.GovUkPay = window.GovUkPay || {};
+    window.GovUkPay.InitAnalytics = {};
+    window.GovUkPay.InitAnalytics.InitialiseAnalytics = jest.fn();
+
+    require("../helpers/cookie/cookie-functions");
+    require("./cookie-settings");
+
+    JsCookie.remove("govuk_pay_cookie_policy");
+
+    document.body.innerHTML = testCookiePageTemplate;
+
+    document.querySelector(".pay-cookie-settings-alert--error").style.display =
+      "none";
+    document.querySelector(
+      ".pay-cookie-settings-alert--success"
+    ).style.display = "none";
+  });
+
+  describe("initialising the component", () => {
+    it("hides the cookie banner if present", () => {
+      window.GovUkPay.CookieSettings.checkForFormAndInit();
+
+      expect(
```

Comment:
May be have excepts in one line ? Linter could move to one line or we could check if team has convention

Reply:
Will raise separate PR with running the `standard` linter